src/config.rs

```diff
@@ -44,6 +46,119 @@ impl TryFrom<&str> for CompatFlag {
     }
 }
 
+#[derive(Debug, Clone)]
+struct ClassMapping {
+    pat: String,
+    classes: Vec<String>,
+    glob: Option<glob::Pattern>,
+    regex: Option<Regex>,
+}
+
+impl std::fmt::Display for ClassMapping {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{} {}", self.pat, self.classes.join(" "))
+    }
+}
+
+// INVARIANT: one of glob and regex must be Some(_). To ensure the invariant is upheld, we
+// implement our own `Default::default()` which creates a ClassMapping that matches all nodes but
+// adds no classes.
+impl Default for ClassMapping {
+    fn default() -> Self {
+        Self {
+            pat: "*".to_owned(),
+            glob: Some(glob::Pattern::new("*").unwrap()),
+            regex: None,
+            classes: Vec::new(),
+        }
+    }
+}
+
+fn parse_class_mapping(cmspec: &str) -> Result<(&str, Vec<&str>)> {
+    let mut parts = cmspec.split_whitespace();
+    let pat = parts
+        .next()
+        .ok_or(anyhow!("Expected '<Pattern> <classes>'"))?;
+    // unescape leading '*' for glob patterns. Leading '*' needs to be escaped to avoid YAML
+    // parsing issues unless strings are explicitly wrapped in quotes.
+    let pat = if pat.starts_with("\\*") {
+        pat.strip_prefix('\\').unwrap()
+    } else {
+        pat
+    };
+    let classes = parts.collect::<Vec<&str>>();
+    if classes.is_empty() {
+        return Err(anyhow!("No classes mapped for {pat}"));
+    }
+
+    Ok((pat, classes))
+}
+
+fn replace_regex_backrefs(s: &str) -> String {
+    static RE: OnceLock<Regex> = OnceLock::new();
```

Comment:
Note: `OnceLock` is thread-safe, cf. https://doc.rust-lang.org/std/sync/struct.OnceLock.html

Reply:
My knowledge was stuck at `lazy_static`. Seems to be deprecated now.

Reply:
Yeah, this is the successor for `lazy_static` (now in the standard lib), turns out in this case `LazyLock` is sufficient.